File: api/tts.py
from gtts import gTTS
from io import BytesIO

def handler(request):
    text = request.args.get("text", "")

    if not text:
        return {
            "status": 400,
            "body": "Text is required"
        }

    mp3_bytes = BytesIO()
    tts = gTTS(text=text, lang="id")
    tts.write_to_fp(mp3_bytes)
    mp3_bytes.seek(0)

    return {
        "status": 200,
        "headers": {
            "Content-Type": "audio/mpeg",
            "Content-Disposition": "attachment; filename=tts.mp3"
        },
        "body": mp3_bytes.read()
    }

File: api/tts.js
import { TextToSpeechClient } from "@google-cloud/text-to-speech";

export default async function handler(req, res) {
    const text = req.query.text || "";

    if (!text) {
        return res.status(400).send("Error: text is required.");
    }

    const client = new TextToSpeechClient({
        credentials: JSON.parse(process.env.GOOGLE_TTS_KEY)
    });

    const [response] = await client.synthesizeSpeech({
        input: { text },
        voice: { languageCode: "id-ID", name: "id-ID-Wavenet-A" },
        audioConfig: { audioEncoding: "MP3" }
    });

    res.setHeader("Content-Type", "audio/mpeg");
    res.send(Buffer.from(response.audioContent, "base64"));
}
